streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -620,22 +620,35 @@ public final TopologyBuilder copartitionSources(Collection<String> sourceNodes)
      * Returns the copartition groups.
      * A copartition group is a group of source topics that are required to be copartitioned.
      *
+     * @param applicationId
      * @return groups of topic names
      */
-    public Collection<Set<String>> copartitionGroups() {
+    public Collection<Set<String>> copartitionGroups(final String applicationId) {
```

Comment:
Yeah that is true; on the other hand since `copartitionGroups` is a public API, I'm a bit worried that requiring the application id here would confuse users who work directly on the Processor layer. So I'd still prefer the library to handle getting the application id upon constructing the builder instead of requiring that from users.